peterpy/handlers/products.py

```diff
@@ -5,41 +5,27 @@
 from aiohttp.web import Request, Response, json_response
 
 from peterpy import routes
+from peterpy.database.connection import DatabaseConnection, DatabaseSession
+
 from peterpy.entities import ProductEncoder
-from peterpy.repositories import MemoryProductRepository
+from peterpy.entities.product import Product
+from peterpy.interfaces.repository import IRepository
+from peterpy.repositories import DatabaseProductRepository
+from peterpy.services import ProductService
 from peterpy.services import ProductService, KafkaService
 
-# define globally because we are using a memory repository
-# every method of the service will use the same repository
-product_repository = MemoryProductRepository()
-
-
-@routes.post("/message")
-async def send_message(request: Request) -> Response:
-    logging.debug("---------------------------------")
-    logging.info("Send message requested from %s", request.remote)
-
-    data = await request.json()
-    topic = "messages"
-    message = data.get("message")
-
-    kafka_service = KafkaService(topic)
-    await kafka_service.produce(message)
-
-    return json_response(status=201, text=json.dumps({"message": "Message sent"}))
-
 
 @routes.get("/list")
 async def list_products(request: Request) -> Response:
     logging.debug("---------------------------------")
     logging.info("List products requested from %s", request.remote)
 
-    product_service = ProductService(product_repository)
+    product_service: ProductService = request["product_service"]
+
     products = product_service.all()
-    products_count = product_service.count()
 
     product_json = json.dumps(
-        {"products": products, "count": products_count}, cls=ProductEncoder
+        {"products": products, "count": len(products)}, cls=ProductEncoder
```

Comment:
Let's discuss pagination in our next meeting :)

Reply:
yeah I removed the query for count here because it would be a not needed query in this case without pagination.
But with pagination, count should be a separate query to get the full count, and use LIMIT and OFFSET to get a paginated set of records